### PR TITLE
feat(query): Support use parquet format when spilling

### PR DESCRIPTION
**Pull request by forsaken628**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
Support use `parquet` format when spilling, you can switch to arrow ipc via `set spilling_file_format = 'arrow'`.

## Tests

- [x] Unit Test
- [ ] Logic Test
- [x] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16612)

### CODE REVIEW

**Comment by forsaken628** [comment]
Benchmark:

dataset: tpch sf100

settings:
```
set max_memory_usage = 16*1024*1024*1024;
set window_partition_spilling_memory_ratio = 30;
set window_partition_spilling_to_disk_bytes_limit = 30*1024*1024*1024;
```

sql
```
EXPLAIN ANALYZE SELECT
    l_orderkey,
    l_partkey,
    l_quantity,
    l_extendedprice,
    l_shipinstruct,
    l_shipmode,
    ROW_NUMBER() OVER (PARTITION BY l_orderkey ORDER BY l_extendedprice DESC) AS row_num,
    RANK() OVER (PARTITION BY l_orderkey ORDER BY l_extendedprice DESC) AS rank_num
FROM
    lineitem ignore_result;
```

```
set spilling_use_parquet = 0; 

        ├── estimated rows: 600037902.00
        ├── cpu time: 651.285131424s
        ├── wait time: 168.630024827s
        ├── output rows: 600.04 million
        ├── output bytes: 44.87 GiB

        ├── numbers local spilled by write: 208
        ├── bytes local spilled by write: 15.06 GiB
        ├── local spilled time by write: 136.856s

        ├── numbers local spilled by read: 3072
        ├── bytes local spilled by read: 15.06 GiB
        ├── local spilled time by read: 31.933s
```

```
set spilling_use_parquet = 1; 

        ├── estimated rows: 600037902.00
        ├── cpu time: 848.406496078s
        ├── wait time: 73.858260885s
        ├── output rows: 600.04 million
        ├── output bytes: 44.87 GiB

        ├── numbers local spilled by write: 208
        ├── bytes local spilled by write: 9.56 GiB
        ├── local spilled time by write: 55.665s

        ├── numbers local spilled by read: 3072
        ├── bytes local spilled by read: 9.56 GiB
        ├── local spilled time by read: 17.512s
```

Compared with arrow ipc, the optimization of parquet's file size mainly comes from dictionary encoding. parquet's cpu usage is quite high at the same time. There is no significant advantage for highly discrete data.

File: src/query/settings/src/settings_default.rs [hunk 1]
```diff
@@ -309,6 +309,12 @@ impl DefaultSettings {
                     mode: SettingMode::Both,
                     range: Some(SettingRange::Numeric(0..=u64::MAX)),
                 }),
+                ("spilling_use_parquet", DefaultSettingValue {
```
Comment (sundy-li) [inline src/query/settings/src/settings_default.rs:312]: let's rename it to `spilling_data_format` which ranges from [  "arrow, "parquet" ]

**Review by Dousir9 — APPROVED** [review]
LGTM !

**Comment by sundy-li** [comment]
LGTM, need rebase.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**